### DISCUSSION TITLE
La corbeille ne sert à rien...

### DISCUSSION THREAD

**Discussion by Tournesol01** [A-S-T-U-C-E/STudio4Education#47] [Ideas]
ST4 bouscule ces standards avec sa barre d'outils originale et très fournie.

Pourquoi donc ne pas aller au delà ? Au diable la conformité d'aspect et d'ergonomie avec les autres blockly !

Voici quelques suggestions qui n'engagent que moi.

LA CORBEILLE NE SERT À RIEN : 

- elle encombre l'espace de travail ;
- quand on travaille sur un sketch bien fourni, il arrive très très souvent qu'elle soit masquée sous des blocs, ce qui rend son accès impossible ;

Or il suffit tout simplement d'utiliser la touche «Suppr.» pour supprimer le bloc ou le groupe de blocs sélectionné. C'est un standard de tous les OS.
⇒ c'est plus efficace, plus rapide, plus sûr ;
⇒ on a de toute façon l'autre possibilité d'effacer : le menu contextuel avec «Delete x blocs» ;

⇒ une autre façon d'effacer un bloc ou un groupe de blocs existe déjà : faire glisser le groupe de blocs sur la «Tool Box» pour rendre à la «Tool Box» les blocs qu'on y a puisés et qu'on ne veut plus : intuitif et logique (c'est la première manip que mes petits enfants ont naturellement utilisée ! ).
Il suffirait juste, dans ce cas, que la zone de la «Tool Box» se transforme en corbeille (en filigrane, comme dans «mBlock» ).

(A noter que cette dernière méthode fonctionne parfaitement après qu'on soit passé en mode plein écran ou qu'on en soit sorti).

En conclusion : pourquoi s'encombrer de 4 manières d'effacer des blocs, alors que 2 suffisent ET fonctionnent correctement ?

**Comment by SebCanet** [comment]
La corbeille permet de récupérer ce qui a été effacé à n'importe quel moment, et non pas ordre chronologique du Ctrl-Z, donc il lui faudrait un palliatif qui n'existe pas.
Par ailleurs, il faut bien avoir en tête que le projet se destine aux débutants et experts, de tous les âges et la corbeille est dans tous les OS le moyen de jeter quelque chose, les enfants y sont habitués. Plus ç a se rapproche de ce qu'ils connaissent et plus il seront à l'aise, ergonomie implicite, prise en main où le blocage vient de ce qui est nouveau et non pas de l'apprentissage forcé du logiciel.
Bref un chantier entamé est de proposer différentes interfaces en fonction du niveau sélectionné : débutant, confirmé, expert.
